### PR TITLE
Updated history checks

### PR DESCRIPTION
**Pull request by Jasleen1**
Minor update to key history verification to ensure previous versions were retired at appropriate times. Addresses https://github.com/novifinancial/akd/issues/222

### CODE REVIEW

**Review by afterdusk — APPROVED** [review]
Perhaps we can consider adding a test for this path, but otherwise it looks good to me!

**Review by slawlor — APPROVED** [review]
> Perhaps we can consider adding a test for this path, but otherwise it looks good to me!

Totally agree with this, adding a small test coverage and looks good-to-go. Thanks!

**Comment by Jasleen1** [comment]
Added a test for the key history check. Note that the method for testing here is using a modified version of the server's `publish` function, along with an `enum` to represent the specific types of corruptions. This method would likely generalize for testing other kinds of server misbehavior as well.

File: akd/src/directory.rs [hunk 1]
```diff
@@ -872,3 +872,196 @@ pub async fn get_directory_root_hash_and_ep<
...
+
+#[cfg(test)]
+#[derive(Debug, Clone)]
+pub(crate) enum PublishCorruption {
+    /// Indicates to the malicious publish function to not mark a stale version
+    UnmarkedStaleVersion(AkdLabel),
+    /// Indicates to the malicious publish to mark a certain version for a username as stale.
+    MarkVersionStale(AkdLabel, u64),
+    /// No corruption
+    NoCorruption,
```
Comment (kevinlewi) [inline akd/src/directory.rs:886]: @Jasleen1 : Seems like NoCorruption isn't used anywhere. Can we delete this to resolve the warning?

Also, can we move this code into tests.rs?

Comment (kevinlewi) [reply akd/src/directory.rs]: Discussed offline with Jasleen, it should be ok to keep in directory.rs. Let's land this!

**Comment by kevinlewi** [comment]
LGTM, thanks!

**Review by kevinlewi — APPROVED** [review]

**Pull request merged**